src/rgw/rgw_notify.cc

```diff
@@ -815,86 +825,102 @@ int publish_reserve(const DoutPrefixProvider *dpp, EventType event_type,
 }
 
 int publish_commit(rgw::sal::Object* obj,
-        uint64_t size,
-        const ceph::real_time& mtime, 
-        const std::string& etag, 
-        const std::string& version,
-        EventType event_type,
-        reservation_t& res,
-        const DoutPrefixProvider *dpp) 
+		   uint64_t size,
+		   const ceph::real_time& mtime,
+		   const std::string& etag,
+		   const std::string& version,
+		   EventType event_type,
+		   reservation_t& res,
+		   const DoutPrefixProvider* dpp)
 {
   for (auto& topic : res.topics) {
-    if (topic.cfg.dest.persistent && topic.res_id == cls_2pc_reservation::NO_ID) {
+    if (topic.cfg.dest.persistent &&
+	topic.res_id == cls_2pc_reservation::NO_ID) {
       // nothing to commit or already committed/aborted
       continue;
     }
     event_entry_t event_entry;
-    populate_event_from_request(res, obj, size, mtime, etag, version, event_type, event_entry.event);
+    populate_event(res, obj, size, mtime, etag, version, event_type, event_entry.event);
     event_entry.event.configurationId = topic.configurationId;
     event_entry.event.opaque_data = topic.cfg.opaque_data;
     if (topic.cfg.dest.persistent) { 
       event_entry.push_endpoint = std::move(topic.cfg.dest.push_endpoint);
-      event_entry.push_endpoint_args = std::move(topic.cfg.dest.push_endpoint_args);
+      event_entry.push_endpoint_args =
+	std::move(topic.cfg.dest.push_endpoint_args);
       event_entry.arn_topic = std::move(topic.cfg.dest.arn_topic);
```

Comment:
```suggestion
      event_entry.arn_topic = topic.cfg.dest.arn_topic;
```
this is an existing bug of "use after move". not sure how it was working before :-(
but now it is failing the persistent notifications test

Reply:
ouch, thanks, will change